src/OrleansProviders/Storage/MemoryStorage.cs

```diff
@@ -108,7 +108,7 @@ public virtual Task Init(string name, IProviderRuntime providerRuntime, IProvide
             for (int i = 0; i < numStorageGrains; i++)
             {
                 int idx = i; // Capture variable to avoid modified closure error
-                storageGrains[idx] = new Lazy<IMemoryStorageGrain>(() => MemoryStorageGrainFactory.GetGrain(idx));
+                storageGrains[idx] = new Lazy<IMemoryStorageGrain>(() => providerRuntime.GrainFactory.GetGrain<IMemoryStorageGrain>(idx));
```

Comment:
I think it'll be cleaner to inject it via IProviderRuntime, just like we do with IGrainRuntime. IProvider is what providers implement, so I think it would be odd to have them 'implement' GrainFactory.